gnustep-base/gnustep-base fails with CFLAGS="-march=native -O3 ${SAFEST_FAST_MATH} ${GRAPHITE} ${DEVIRTLTO} ${IPAPTA} ${SEMINTERPOS} ${FLTO} -fuse-linker-plugin"

Unicode.m: In function ‘GSSetupEncodingTable.part.0’:
Unicode.m:339:9: note: length computed here
In function ‘strncpy’,
    inlined from ‘newDataWithEncodedPort’ at NSMessagePort.m:199:3:
/usr/include/bits/string_fortified.h:106:10: warning: ‘__builtin_strncpy’ specified bound depends on the length of the source argument [-Wstringop-overflow=]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^
NSMessagePort.m: In function ‘newDataWithEncodedPort’:
NSMessagePort.m:199:41: note: length computed here
  199 |   strncpy((char*)pi->addr, (char*)name, strlen((char*)name) + 1);
      |                                         ^
/var/tmp/portage/gnustep-base/gnustep-base-1.27.0/temp/ccf282Pz.s: Assembler messages:
/var/tmp/portage/gnustep-base/gnustep-base-1.27.0/temp/ccf282Pz.s:17929: Error: symbol `__objc_class_ref__GSMutableInsensitiveDictionary.lto_priv.1' is already defined
make[5]: *** [/var/tmp/portage/gnustep-base/gnustep-base-1.27.0/temp/cc6CRVb8.mk:218: /var/tmp/portage/gnustep-base/gnustep-base-1.27.0/temp/libgnustep-base.so.1.27.0.OxSAn7.ltrans72.ltrans.o] Error 1
lto-wrapper: fatal error: make returned 2 exit status
compilation terminated.
/usr/lib/gcc/x86_64-pc-linux-gnu/10.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
gmake[4]: *** [/usr/share/GNUstep/Makefiles/Instance/library.make:293: obj/libgnustep-base.so.1.27.0] Error 1
gmake[3]: *** [/usr/share/GNUstep/Makefiles/Instance/library.make:278: internal-library-all_] Error 2
gmake[2]: *** [/usr/share/GNUstep/Makefiles/Master/rules.make:297: libgnustep-base.all.library.variables] Error 2
gmake[1]: *** [/usr/share/GNUstep/Makefiles/Master/library.make:37: internal-all] Error 2
gmake[1]: Leaving directory '/var/tmp/portage/gnustep-base/gnustep-base-1.27.0/work/gnustep-base-1.27.0/Source'
make: *** [/usr/share/GNUstep/Makefiles/Master/serial-subdirectories.make:53: internal-all] Error 2

--- a/sys-config/ltoize/files/package.cflags/ltoworkarounds.conf
+++ b/sys-config/ltoize/files/package.cflags/ltoworkarounds.conf
@@ -109,6 +109,7 @@ app-text/yodl *FLAGS-=-flto* # Fixes build
 sys-devel/llvm *FLAGS-=-flto* # Issue #619 temporarily disabled for now due to build errors
 sys-devel/clang *FLAGS-=-flto* # Issue #619 Same as above
 sys-libs/libomp *FLAGS-=-flto* # Issue #619 Same as above
+gnustep-base/gnustep-make *FLAGS-=-flto* # Issue #581, tools to build gnustep-base/gnustep-base, if built with LTO doesn't build gnustep code
 
 # BEGIN: LTO not recommended
 # Packages which can be built with LTO but leads to problems/crashes/segfaults

